source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRLongPressGestureRecognizerEventHandler.mm

```diff
@@ -0,0 +1,59 @@
+//
+//  ACRLongPressGestureRecognizerEventHandler
+//  ACRLongPressGestureRecognizerEventHandler.mm
+//
+//  Copyright © 2018 Microsoft. All rights reserved.
+//
+
+#import "ACRLongPressGestureRecognizerEventHandler.h"
+
+@implementation ACRLongPressGestureRecognizerEventHandler
+{
+    // backgroundColor is one of the cues that provides
+    // users that action is triggered
+    UIColor *_backgroundColor;
+}
+
+- (instancetype)init
+{
+    self = [super init];
+    if(self)
+    {
+        _backgroundColor = nil;
+    }
+    return self;
+}
+// this method does the followings
+// 1. it provides users with cue that select action is about to be initiated
+// 2. execute select action by calling its delegate
+- (IBAction)processLongPressGesture:(UILongPressGestureRecognizer *) recognizer;
+{
+    // activate it when fingers lifts off
+    if(recognizer.state == UIGestureRecognizerStateBegan)
+    {
+        _backgroundColor = recognizer.view.backgroundColor;
```

Comment:
>_backgroundColor [](start = 8, length = 16)

Does this have to be an instance field? Can it just be a local variable?

Reply:
Unfortunately, iOS doesn't have a default style for UIVIew object that's selected. UIButtons do have it.

---
In reply to: [164587016](https://github.com/Microsoft/AdaptiveCards/pull/1075#discussion_r164587016) [](ancestors = 164587016)

Reply:
I don't understand this answer. Why does `_backgroundColor` have to be a field rather than a local variable?

---
In reply to: [164608055](https://github.com/Microsoft/AdaptiveCards/pull/1075#discussion_r164608055) [](ancestors = 164608055,164587016)

Reply:
good point. yes, you are correct. i had earlier version where state changes in two calls to this method. so that required the instance variable. Then i changed to UIViewPropertyAnimator, so it is no longer needed.
Sorry, I didn't pay attention. thanks

---
In reply to: [164823115](https://github.com/Microsoft/AdaptiveCards/pull/1075#discussion_r164823115) [](ancestors = 164823115,164608055,164587016)